lms/djangoapps/certificates/management/commands/generate_persistent_grade.py

```diff
@@ -0,0 +1,38 @@
+"""
+Management command which update/generate persistent grades
+"""
+import logging
+
+from django.core.management.base import BaseCommand
+from django.http import Http404
+from lms.djangoapps.grades.new.course_grade import CourseGradeFactory
+
+from course_blocks.api import get_course_blocks
+from courseware import courses
+from student.models import User, CourseEnrollment
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Management command which update/generate persistent grades
+    """
+
+    help = """
+    Update/generate persistent grades
+    """
+
+    def handle(self, *args, **options):
+        course_structure = None
+        for enrollment in CourseEnrollment.objects.all():
+            user = enrollment.user
+            try:
+                course = courses.get_course_by_id(enrollment.course_id)
+                course_structure = get_course_blocks(user, course.location)
+            except:
+                # Used for catch exceptions related to bad courses
```

Comment:
What exception are we expecting for?

Reply:
The exception can occur when the student for some reason lost access to the course. In normal flow it is impossible, but in some cases, staff can restrict access to the course for the user. Or another case - course can be removed by sysadmin dashboard, but de facto it exists in the system.